File: JavaApp/src/main/java/com/charlezz/javaapp/feature/local/PersonFragment.java
package com.charlezz.javaapp.feature.local;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.charlezz.javaapp.databinding.FragmentPersonBinding;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;

public class PersonFragment extends DaggerFragment {
    @Inject
    FragmentPersonBinding binding;

    @Inject
    PersonViewModel viewModel;

    @Inject
    PersonAdapter adapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.setLifecycleOwner(this);
        binding.recyclerView.setAdapter(adapter);
        binding.setViewmodel(viewModel);
    }
}

File: app/src/main/java/com/charlezz/finalarchitecture/ui/local/PersonFragment.kt
package com.charlezz.finalarchitecture.ui.local

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.charlezz.finalarchitecture.databinding.FragmentPersonBinding
import com.charlezz.finalarchitecture.viewmodel.PersonFragmentViewModel
import dagger.android.support.DaggerFragment
import javax.inject.Inject

class PersonFragment : DaggerFragment() {
    @Inject
    lateinit var binding: FragmentPersonBinding

    @Inject
    lateinit var viewmodel: PersonFragmentViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) = binding.root

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.setLifecycleOwner(this)
        binding.viewmodel = viewmodel
    }

}